### PR TITLE
Segmentation map in Resizing layer

### LINKED ISSUE #1830: Support Segmentation Maps in Resizing Layer
It would be really helpful is the `Resizing` layer supports segmentation maps.

### PR DESCRIPTION
**Pull request by cosmo3769**
# What does this PR do?

Fixes #1830 (issue)

Demo added by @ariG23498  

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [ ] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

@ianstenbit @LukeWood @jbischof

### CODE REVIEW

File: keras_cv/layers/preprocessing/resizing.py [hunk 1]
```diff
@@ -143,24 +154,37 @@ def _augment(self, inputs):
...
         size = [self.height, self.width]
         images = tf.image.resize(
             images, size=size, method=self._interpolation_method
         )
         images = tf.cast(images, self.compute_dtype)
 
+        if segmentation_maps is not None:
+            segmentation_maps = tf.image.resize(
+                segmentation_maps, size=size, method=self._interpolation_method
+            )
```
Comment (ariG23498) [inline keras_cv/layers/preprocessing/resizing.py:176]: I think we should have a way to input different resizing methods of the image and the segmentation masks.

WDYT? @ianstenbit

Comment (ayulockin) [reply keras_cv/layers/preprocessing/resizing.py:176]: Usually segmentation masks are resized with nearest neighbor interpolation in practice. Using say bilinear interpolation ruin the masks.

File: keras_cv/layers/preprocessing/resizing.py [hunk 2]
```diff
@@ -24,7 +24,13 @@
 H_AXIS = -3
 W_AXIS = -2
 
-supported_keys = ["images", "labels", "targets", "bounding_boxes"]
+supported_keys = [
+    "images",
+    "labels",
+    "targets",
+    "bounding_boxes",
+    "segmentation_maps",
```
Comment (ianstenbit) [inline keras_cv/layers/preprocessing/resizing.py:32]: This should be `"segmentation_masks"` for consistency with the other KPLs

Comment (ianstenbit) [reply keras_cv/layers/preprocessing/resizing.py:32]: (Same throughout -- s/maps/masks)

File: keras_cv/layers/preprocessing/resizing.py (hunk 1 above)
Comment (ianstenbit) [reply keras_cv/layers/preprocessing/resizing.py:176]: Yes, this should be resized with nearest neighbor. Numerical interpolation is not safe for masks unless they're one-hot encoded. (See RandomRotation for an example of where we had to deal with this).

For resizing I think it's probably reasonable to just always use nearest neighbor, regardless of whether masks are one-hot encoded.

**Review by ianstenbit — CHANGES_REQUESTED** [review]
Thanks for the PR!

Please also update the resizing test to include a mask resizing, and also please update `with_segmentation_masks_test.py`

Thank you!

File: keras_cv/layers/preprocessing/resizing.py [hunk 3]
```diff
@@ -255,24 +302,53 @@ def _resize_with_crop(self, inputs):
         else:
             input_dtype = tf.float32
 
-        def resize_with_crop_to_aspect(x):
+        def resize_with_crop_to_aspect_images(x):
```
Comment (ianstenbit) [inline keras_cv/layers/preprocessing/resizing.py:305]: Instead of making two versions of this method, let's just add `interpolation_method` as a parameter, and call the same method for both masks and images, just with different parameters.

**Review by ianstenbit — COMMENTED** [review]
Thanks! 
Looking closer -- just need to add tests, and let's also add an example in 
`keras-cv/examples/layers/preprocessing/segmentation/resize_demo.py` (and please include the output of the demo in the PR description.

**Comment by cosmo3769** [comment]
Apologies for the delayed work. I am busy with my exams now. I will resolve the issues and changes requested on 8 June.

File: keras_cv/layers/preprocessing/resizing.py (hunk 3 above)
Comment (ariG23498) [reply keras_cv/layers/preprocessing/resizing.py]: I have used the `functool.partial` to take care of this. Let me know if you want it to be implemented in any other way.

File: keras_cv/layers/preprocessing/resizing.py [hunk 4]
```diff
@@ -255,24 +308,58 @@ def _resize_with_crop(self, inputs):
         else:
             input_dtype = tf.float32
 
-        def resize_with_crop_to_aspect(x):
+        def resize_with_crop_to_aspect_images(x, interpolation_method):
```
Comment (ianstenbit) [inline keras_cv/layers/preprocessing/resizing.py:311]: nit: let's keep the name the way it was (without "_images")

File: keras_cv/layers/preprocessing/resizing.py [hunk 5]
```diff
@@ -255,24 +308,58 @@ def _resize_with_crop(self, inputs):
...
+                x,
+                size=size,
+                interpolation=interpolation_method,
             )
 
         if isinstance(images, tf.RaggedTensor):
             size_as_shape = tf.TensorShape(size)
             shape = size_as_shape + images.shape[-1:]
             spec = tf.TensorSpec(shape, input_dtype)
+            map_fn = partial(
```
Comment (ianstenbit) [inline keras_cv/layers/preprocessing/resizing.py:324]: Ahh I didn't realize we'd end up having to use a `partial` here -- my bad.

How about we keep a base method `resize_with_crop_to_aspect` and make two wrapper methods with the same name and `_images` and `_masks` suffixes which pass through the interpolation method (just to avoid using a partial)

**Review by ianstenbit — APPROVED** [review]
Thanks for the PR! Just a few small comments left
